export/export.go

```diff
@@ -0,0 +1,438 @@
+	errgrp.listPvs(ctx, &s.pvService, &resources)
+	errgrp.listPvcs(ctx, &s.pvcService, &resources)
+	errgrp.listStorageClasses(ctx, &s.storageClassService, &resources)
+	errgrp.listPcs(ctx, &s.priorityclassService, &resources)
+	errgrp.getSchedulerConfig(ctx, &s.schedulerService, &resources)
+
+	if err := errgrp.g.Wait(); err != nil {
+		return nil, xerrors.Errorf("get resources all: %w", err)
+	}
+	return &resources, nil
+}
+
+func (s *Service) Export(ctx context.Context) (*ResourcesForExport, error) {
+	resources, err := s.get(ctx)
+	if err != nil {
+		return nil, xerrors.Errorf("export resources all: %w", err)
+	}
+	return resources, nil
+}
+
+// Import all resources from posted data.
+// (1) Restart scheduler based on the data.
+// (2) Apply each resource to the scheduler.
+//     * If UID is not nil, an error will occur. (try to find existing resource by UID)
+// (3) Get all resources. (Separated the get function to unify the struct format.)
+func (s *Service) Import(ctx context.Context, resources *ResourcesForImport) error {
+	if err := s.schedulerService.RestartScheduler(resources.SchedulerConfig); err != nil {
+		return xerrors.Errorf("restart scheduler with imported configuration: %w", err)
+	}
+	errgrp := NewErrorGroupWithSemaphore(ctx)
+	errgrp.applyPcs(ctx, &s.priorityclassService, resources)
+	errgrp.applyStorageClasses(ctx, &s.storageClassService, resources)
+	s.applyPvsAndPvcs(ctx, resources, &errgrp)
+	errgrp.applyNodes(ctx, &s.nodeService, resources)
+	errgrp.applyPods(ctx, &s.podService, resources)
+
+	if err := errgrp.g.Wait(); err != nil {
+		return xerrors.Errorf("apply each resources: %w", err)
+	}
+	return nil
+}
+
+// The apply of PersistentVolume will be called after the apply of PersistentVolumeClaim is finish.
+// This is because that PersistentVolume search PersistentVolumeClaim and associate itself if they had relationship.
+func (s *Service) applyPvsAndPvcs(ctx context.Context, resources *ResourcesForImport, errgrp *ErrorGroupWithSemaphore) {
+	eg := NewErrorGroupWithSemaphore(ctx)
+	if err := eg.sem.Acquire(ctx, 1); err != nil {
+		klog.Fatalf("failed to acquire semaphore: %v", err)
+		return
+	}
+	eg.g.Go(func() error {
+		defer eg.sem.Release(1)
+		errgrp.applyPvcs(ctx, &s.pvcService, resources)
+		return nil
+	})
+	if err := eg.g.Wait(); err != nil {
+		klog.Fatalf("failed to apply pvcs: %v", err.Error())
+	}
+
+	if err := eg.sem.Acquire(ctx, 1); err != nil {
+		klog.Fatalf("failed to acquire semaphore: %v", err)
+		return
+	}
+	eg.g.Go(func() error {
+		defer eg.sem.Release(1)
+		errgrp.applyPvs(ctx, &s.pvService, &s.pvcService, resources)
+		return nil
+	})
+	if err := eg.g.Wait(); err != nil {
+		klog.Fatalf("failed to apply pvcs: %v", err.Error())
+	}
+}
+
+func (eg *ErrorGroupWithSemaphore) listPods(ctx context.Context, s *PodService, r *ResourcesForExport) {
+	if err := eg.sem.Acquire(ctx, 1); err != nil {
+		klog.Fatalf("failed to acquire semaphore: %v", err)
```

Comment:
return error instead of `klog.Fatalf`. `klog.Fatalf` calls `os.Exit` internally. Other places as well.

`os.Exit` shouldn't be used in most cases. (= `klog.Fatalf` also shouldn't be used)